src/lib.rs

```diff
@@ -47,16 +47,28 @@ use serde::ser::Serializer;
 #[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
 
+const SHARD_COUNT: usize = 16;
+
 /// The global string cache for `FlyStr`.
 ///
 /// If a live `FlyStr` contains an `Storage`, the `Storage` must also be in this cache and it must
 /// have a refcount of >= 2.
-static CACHE: std::sync::LazyLock<Cache> = std::sync::LazyLock::new(Cache::default);
+static CACHE: std::sync::LazyLock<Cache> = std::sync::LazyLock::new(|| Cache {
+    hasher: RandomState::default(),
+    tables: std::array::from_fn(|_| Default::default()),
+});
 
-#[derive(Default)]
 struct Cache {
     hasher: RandomState,
-    table: Mutex<HashTable<Storage>>,
+    tables: [Mutex<HashTable<Storage>>; SHARD_COUNT],
```

Comment:
Maybe we should [pad the mutex to cache lines](https://docs.rs/crossbeam/latest/crossbeam/utils/struct.CachePadded.html) to avoid contention?